app/scripts/models/ThreeViewer.js

```diff
@@ -0,0 +1,306 @@
+            partMat,
+            lineMat,
+            mesh,
+            geometry,
+            count,
+            size,
+            positions,
+            index = 0,
+            pos = 0;
+
+        if (selected) {
+            color = colorLine = colors.selected;
+        }
+        else {
+            color = colors.notSelected;
+            colorLine = colors.notSelectedLine;
+        }
+
+        meshMat = new THREE.MeshPhongMaterial({ color: color, side: THREE.DoubleSide });
+        partMat = new THREE.PointCloudMaterial({ color: color, size: 5, sizeAttenuation: false, side: THREE.DoubleSide });
+        lineMat = new THREE.LineBasicMaterial({ color: colorLine });
+
+        geometry = new THREE.BufferGeometry();
+
+        if (list.lineStripVertices.length && list.lineStripCounts.length) {
+
+            for (var k = 0; k < list.lineStripCounts.length; k++) {
+
+                positions = [];
+                count = list.lineStripCounts[k];
+                if (!count) {
+                    continue;
+                }
+
+                size = index + count;
+                for (var i = index; i < size; i++) {
+                    pos = i * 3;
+                    positions.push(
+                        list.lineStripVertices[pos],
+                        list.lineStripVertices[pos + 1],
+                        list.lineStripVertices[pos + 2]
+                    );
+                }
+
+                index += count;
+                geometry.addAttribute('position', new THREE.BufferAttribute(new Float32Array(positions), 3));
+            }
+
+            geom.add(new THREE.Line(geometry, lineMat, THREE.LinePieces));
+        }
+
+        if (list.pointVertices.length) {
+
+            geometry.addAttribute('position', new THREE.BufferAttribute(list.pointVertices, 3));
+            mesh = new THREE.PointCloud(geometry, partMat);
+
+            geom.add(mesh);
+        }
+
+        if (list.triangleVertices.length) {
+
+            geometry.addAttribute('position', new THREE.BufferAttribute(list.triangleVertices, 3));
+            geometry.addAttribute('normal', new THREE.BufferAttribute(list.triangleNormals, 3));
+            mesh = new THREE.Mesh(geometry, meshMat);
+
+            geom.add(mesh);
+        }
+    };
+
+    function onRequestGeometry(){
+        if(this.pendingRequestsCount === 0)
+            this.app.trigger('show-progress');
+        this.pendingRequestsCount++;
+    };
+
+    function zoomToFit(){
+        this.viewPort.zoomToFit();
+    };
+
+    function dollyOut(){
+        this.viewPort.dollyOut();
+    };
+
+    function dollyIn(){
+        this.viewPort.dollyIn();
+    };
+
+    function exportSTL(e){
+        GeometryExport.toSTL(this.viewPort.scene, e.name);
+    };
+
+    return Backbone.Model.extend({
+
+        initialize: function (args, options) {
+            this.viewPort = new Viewport();
+            this.viewPort.init();
+
+            this.app = args.app;
+            this.idMap = {};
```

Comment:
Here we are saving link to threeGeometry with node Id, so we can easily manipulate this data.